grpc-zpages/docker/envoy/zprox.sh

```diff
@@ -56,6 +58,27 @@ static_resources:
                 route:
                   cluster: local_service_web
           http_filters:
+          - name: envoy.lua
+            config:
+              inline_code: |
+                function envoy_on_request(request_handle)
+                  -- hyphens are special characters in lua search patterns and must be escaped with a %
+                  function esc_hyphens(s)
+                    result, _ = string.gsub(s, "%-", "%%%-")
+                    return result
+                  end
+                  headers = request_handle:headers()
+                  content_type = headers:get("content-type")
+                  if content_type and string.find(content_type, esc_hyphens("application/grpc-web")) then
+                    xsrf_header = headers:get("$GDEBUG_TOKEN_NAME")
+                    xsrf_cookie = string.match(headers:get("cookie") or "", esc_hyphens("$GDEBUG_TOKEN_NAME=(%w+)"))
```

Comment:
nit: I think you need to parse the whole cookie, because someone could embed the k=v string inside of their own cookie name I think.

Reply:
You're right. My browser seems to enforce that semicolon can not appear in the value, so using `; ` and `^` as criteria are probably sufficient.